### PR TITLE
Create lockfile to assert single instance execution of kytosd

### LINKED ISSUE #340: Abort when another instance is running
Today, if one instance is already started we should abort.

There is a try exception but is not working since that we are working with threads.

### PR DESCRIPTION
**Pull request by renanrodrigo**
Fix #340

### CODE REVIEW

**Comment by renanrodrigo** [comment]
Depends on #361

File: kytos/core/controller.py [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
 __all__ = ('Controller',)
 
+LOCK_PATH = '/tmp/lockytos'
```
Comment (beraldoleal) [inline kytos/core/controller.py]: This should be configured on `/etc/kytos/kytos.conf`. And should have a default value to `/var/run/kytos.pid`

File: kytos/core/controller.py [hunk 2]
```diff
@@ -116,6 +118,34 @@ def enable_logs(self):
         self.log = logging.getLogger(__name__)
 
     def start(self):
+        """Create lockfile and call start_controller method."""
+        pid = os.getpid()
+
+        # Checks if a lockfile exists. Creates a new file.
+        try:
+            lockfile = open(LOCK_PATH, mode='x')
```
Comment (beraldoleal) [inline kytos/core/controller.py]: s/lockfile/pidfile/g

File: kytos/core/controller.py [hunk 3]
```diff
@@ -116,6 +118,34 @@ def enable_logs(self):
...
+            # running.
+            try:
+                existing_file = open(LOCK_PATH, mode='r')
+                old_pid = int(existing_file.read())
+                os.kill(old_pid, 0)
+                # If kill() doesn't returns an error, it is still running.
+                # Otherwise, overwrite the file and proceed.
+                self.log.info("Failed to create a lockfile."
+                              "Is kytos running?")
+                exit(1)
```
Comment (beraldoleal) [inline kytos/core/controller.py]: Replace 1 with a proper constant. For instance `os.EX_CANTCREAT`

File: kytos/core/controller.py [hunk 4]
```diff
@@ -212,6 +243,7 @@ def stop_controller(self, graceful=True):
         self.started_at = None
         self.unload_napps()
         self.buffers = KytosBuffers()
+        os.remove(LOCK_PATH)
```
Comment (beraldoleal) [inline kytos/core/controller.py]: s/LOCK_PATH/PID_FILE/g.

Also you have to handle exceptions like `FileNotFoundError`

**Review by beraldoleal — CHANGES_REQUESTED** [review]

**Comment by renanrodrigo** [comment]
Rebased and working

**Pull request merged**